src/main/java/com/getindata/connectors/http/sink/httpclient/JavaNetSinkHttpClient.java

```diff
@@ -0,0 +1,111 @@
+package com.getindata.connectors.http.sink.httpclient;
+
+import com.getindata.connectors.http.SinkHttpClient;
+import com.getindata.connectors.http.SinkHttpClientResponse;
+import com.getindata.connectors.http.sink.HttpSink;
+import com.getindata.connectors.http.sink.HttpSinkRequestEntry;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * An implementation of {@link SinkHttpClient} that uses Java 11's {@link HttpClient}.
+ */
+public class JavaNetSinkHttpClient implements SinkHttpClient {
+  private final HttpClient httpClient;
+
+  public JavaNetSinkHttpClient() {
+    this.httpClient = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
+  }
+
+  /**
+   * A wrapper structure around an arbitrary element, keeping a reference to a particular
+   * {@link HttpSinkRequestEntry}. Used internally by the {@code HttpSinkWriter} to pass
+   * {@code HttpSinkRequestEntry} along some other element that is logically connected with it
+   * (e.g., full HTTP request built from the {@code HttpSinkRequestEntry}).
+   *
+   * @param <T>
+   */
+  @RequiredArgsConstructor
+  @EqualsAndHashCode
+  private static class HttpSinkRequestEntryWrapper<T> {
+    /**
+     * An element logically connected with the {@link HttpSinkRequestEntry}.
+     */
+    public final T element;
+
+    /**
+     * A representation of a single {@link HttpSink} request.
+     */
+    public final HttpSinkRequestEntry sinkRequestEntry;
+  }
+
+  @Override
+  public CompletableFuture<SinkHttpClientResponse> putRequests(
+      List<HttpSinkRequestEntry> requestEntries, String endpointUrl
+  ) {
+    var endpointUri = URI.create(endpointUrl);
+
+    List<HttpSinkRequestEntryWrapper<HttpRequest>> requests = requestEntries
+        .stream()
+        .map(requestEntry -> new HttpSinkRequestEntryWrapper<>(
+            buildHttpRequest(requestEntry, endpointUri),
+            requestEntry
+        ))
+        .collect(Collectors.toList());
```

Comment:
Like discussed on chat, lets change this to something like this:

```
List<CompletableFuture<HttpResponse<String>>> results = new ArrayList<>(requestEntries.size());

for (HttpSinkRequestEntry entry : requestEntries) {

  CompletableFuture<HttpResponse<String>> future =
      httpClient.sendAsync(
          buildHttpRequest(entry, endpointUri),
          BodyHandlers.ofString()
      );
  results.add(future);
}
```
Then we can modify `getCompletedFutures` which will made `HttpSinkRequestEntryWrapper` used only for wrapping response.